### ISSUE TITLE
Update CI test matrix

### ISSUE THREAD

**Issue by JohnAlbin** [chapter-three/next-drupal#543]
The CI testing matrix currently (Oct 9, 2023) has:

```
php: ["8.0", "8.1"]
drupal: ["9.4", "9.5", "10.0"]
exclude:
  - php: "8.0"
    drupal: "10.0"
```

D9.4 became unsupported when D10.1 was released. Also, D10.x supports PHP 8.2. So the matrix needs to be updated to:

```
php: ["8.0", "8.1", "8.2"]
drupal: ["9.5", "10.0", "10.1"]
exclude:
  - php: "8.1"
    drupal: "9.5"
  - php: "8.2"
    drupal: "9.5"
  - php: "8.0"
    drupal: "10.0"
  - php: "8.0"
    drupal: "10.1"
```

**Comment by JohnAlbin** [comment]
Any PR for this will depend on the CI tests being fixed first: #541

**Comment by JohnAlbin** [comment]
There are 3 failing tests when adding Drupal 10.1 to the CI matrix. I'm splitting getting 10.1 tests working into its own ticket. #547
